src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -222,7 +230,25 @@ private CommandResult executeCommand(String commandText) throws CommandException
              * and update the UI accordingly, so it is reset instead to reduce the chance of bugs.
              */
             bodyPanel.getAddressPanel().getPersonListPanel().scrollToTop();
-            bodyPanel.getAddressPanel().getPersonListPanel().clearSelection();
         }
     }
+
+    /**
+     * Sets selected person's details in person detail panel.
+     */
+    private void setPersonDetail() {
+        ReadOnlyObjectProperty<Person> observableSelectedPerson = logic.getSelectedPerson();
+        bodyPanel.getAddressPanel().getPersonDetailPanel().setPerson(observableSelectedPerson.getValue());
+    }
+
+    /**
+     * Sets selected person's current index in person detail panel
+     */
+    private void setDisplayedIndex() {
+        ReadOnlyObjectProperty<Index> observableSelectedIndex = logic.getSelectedIndex();
+        bodyPanel.getAddressPanel().getPersonDetailPanel()
+                .setDisplayedIndex(observableSelectedIndex.getValue().getOneBased());
+        bodyPanel.getAddressPanel().getPersonListPanel()
+                .bindSelectedIndex((observableSelectedIndex.getValue().getOneBased()));
+    }
```

Comment:
Actually, instead of having the the chain of getters, I think have a `BodyPanel#setSelectedPerson(Person person)` method is better as follows the law of demeter (don't talk to strangers).

So something like:
```java
logic.getSelectedPerson().addListener((observable, oldValue, newValue) -> {
    bodyPanel.setSelectedPerson(newValue);
});
```